pkg/types/config/cluster.go

```diff
@@ -150,3 +158,114 @@ func (c *Cluster) YAML() (string, error) {
 
 	return string(yaml), nil
 }
+
+// ConvertInstallConfigToTFVar converts the installconfig to the Cluster struct
+// that represents the terraform.tfvar file.
+// TODO(yifan): Clean up the Cluster struct to trim unnecessary fields.
+func ConvertInstallConfigToTFVar(cfg *types.InstallConfig, bootstrapIgn string, masterIgns []string, workerIgn string) (*Cluster, error) {
+	cluster := &Cluster{
+		Admin: Admin{
+			Email:    cfg.Admin.Email,
+			Password: cfg.Admin.Password,
+			SSHKey:   cfg.Admin.SSHKey,
+		},
+
+		IgnitionMasters:   masterIgns,
+		IgnitionWorker:    workerIgn,
+		IgnitionBootstrap: bootstrapIgn,
+
+		Internal: Internal{
+			ClusterID: cfg.ClusterID,
+		},
+
+		Networking: Networking{
+			Type:        tectonicnetwork.NetworkType(cfg.Networking.Type),
+			ServiceCIDR: cfg.Networking.ServiceCIDR.String(),
+			PodCIDR:     cfg.Networking.PodCIDR.String(),
+		},
+		BaseDomain: cfg.BaseDomain,
+		Name:       cfg.Name,
```

Comment:
How does this work with #209 rejected?  Is it using property promotion?  I'd rather use the explicit `cfg.Metadata.Name`.

Reply:
ObjectMeta is an embedded struct in InstallConfig.